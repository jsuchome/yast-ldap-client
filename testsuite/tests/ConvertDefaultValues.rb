# encoding: utf-8

#  ConvertDefaultValues.ycp
#  Summary:	Test of Ldap::ConvertDefaultValues
#  Author:	Jiri Suchomel <jsuchome@suse.cz>
#  $Id$
module Yast
  class ConvertDefaultValuesClient < Client
    def main
      # testedfiles: Ldap.ycp
      Yast.import "Testsuite"

      @READ = { "target" => { "tmpdir" => "/tmp", "stat" => {} } }
      Testsuite.Init([@READ], 0)

      Yast.import "Ldap"

      @template = {
        "suseDefaultValue" => [
          "homeDirectory=/home/%uid",
          "loginShell=/bin/bash",
          "testvalue=first=second",
          "novalue"
        ]
      }
      Testsuite.Test(lambda { Ldap.ConvertDefaultValues(@template) }, [@READ], 0)

      nil
    end
  end
end

Yast::ConvertDefaultValuesClient.new.main
