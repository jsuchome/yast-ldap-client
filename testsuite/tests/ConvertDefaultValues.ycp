/**
 *  ConvertDefaultValues.ycp
 *  Summary:	Test of Ldap::ConvertDefaultValues
 *  Author:	Jiri Suchomel <jsuchome@suse.cz>
 *  $Id$
 */

{
    // testedfiles: Ldap.ycp
    import "Testsuite";
    import "Ldap";

    map template	= $[
	"suseDefaultValue"	: [
	    "homeDirectory=/home/%uid",
	    "loginShell=/bin/bash",
	    "testvalue=first=second",
	    "novalue"
	]
    ];
    Testsuite::Test (``(Ldap::ConvertDefaultValues (template)), [], 0);
}

