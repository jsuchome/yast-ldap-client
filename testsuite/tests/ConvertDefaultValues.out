Return	$["default_values":$["homeDirectory":"/home/%uid", "loginShell":"/bin/bash", "novalue":"", "testvalue":"first=second"], "suseDefaultValue":["homeDirectory=/home/%uid", "loginShell=/bin/bash", "testvalue=first=second", "novalue"]]
